azure-iot-device/tests/provisioning/internal/test_polling_machine.py

```diff
@@ -0,0 +1,720 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import pytest
+import datetime
+from mock import MagicMock
+from azure.iot.device.provisioning.internal.request_response_provider import RequestResponseProvider
+from azure.iot.device.provisioning.internal.polling_machine import PollingMachine
+from azure.iot.device.provisioning.models.registration_result import RegistrationResult
+from azure.iot.device.provisioning.transport import constant
+import time
+
+fake_request_id = "Request1234"
+fake_retry_after = "3"
+fake_operation_id = "Operation4567"
+fake_status = "Flying"
+fake_device_id = "MyNimbus2000"
+fake_assigned_hub = "Dumbledore'sArmy"
+fake_sub_status = "FlyingOnHippogriff"
+fake_created_dttm = datetime.datetime(2020, 5, 17)
+fake_last_update_dttm = datetime.datetime(2020, 10, 17)
+fake_etag = "HighQualityFlyingBroom"
+fake_symmetric_key = "Zm9vYmFy"
+fake_registration_id = "MyPensieve"
+fake_id_scope = "Enchanted0000Ceiling7898"
+fake_success_response_topic = "$dps/registrations/res/200/?"
+fake_failure_response_topic = "$dps/registrations/res/400/?"
+fake_greater_429_response_topic = "$dps/registrations/res/430/?"
+fake_assigning_status = "assigning"
+fake_assigned_status = "assigned"
+
+
+class TestRequestResponseProvider(RequestResponseProvider):
+    def receive_response(self, topic, payload):
+        return super(TestRequestResponseProvider, self)._receive_response(topic, payload)
+
+
+@pytest.fixture
+def mock_request_response_provider(mocker):
+    return mocker.MagicMock(spec=TestRequestResponseProvider)
+
+
+@pytest.fixture
+def mock_polling_machine(mocker, mock_request_response_provider):
+    state_based_mqtt = MagicMock()
+    mock_init_request_response_provider = mocker.patch(
+        "azure.iot.device.provisioning.internal.polling_machine.RequestResponseProvider"
+    )
+    mock_init_request_response_provider.return_value = mock_request_response_provider
+    mock_polling_machine = PollingMachine(state_based_mqtt)
+    return mock_polling_machine
+
+
+class TestRegister:
```

Comment:
Again here and everywhere (in all test files) please use pytest.mark.it and pytest.mark.describe as outlined in the wiki